modules/core/arrow-core-data/src/main/kotlin/arrow/core/Endo.kt

```diff
@@ -0,0 +1,8 @@
+package arrow.core
+
+/**
+ * The monoid of endomorphisms under composition.
+ */
+data class Endo<A>(val f: (A) -> A) {
```

Comment:
Unless there is an instance of Eq for this it would be more efficient represented as just a class not a data class

Reply:
Can we provide some useful use cases that de demonstrates the use of this new data type ?

Reply:
Maybe the easiest use case is function composition, although there is even a [package](http://hackage.haskell.org/package/endo-0.3.0.1) to showcase some useful examples, maybe we can port some of the as docs for Arrow! 😉